ElectronClient/gui/ConfigScreen.jsx

```diff
@@ -281,6 +283,9 @@ class ConfigScreenComponent extends React.Component {
 
 		const updateSettingValue = (key, value) => {
 			// console.info(key + ' = ' + value);
+			if (key == 'themeAutoDetect') {
+				updateSettingValue('shouldUseDarkColors', bridge().shouldUseDarkColors());
```

Comment:
I tried my best to avoid making changes to this component but I think this line required here to detect system theme because `bridge()` is already imported in this component and also Auto Detect feature is specific to ElectronClient so I think we cannot call `shouldUseDarkColors()` from `BaseApplication.js` as `bridge()` cannot be imported in this file as it's  common for all Clients.
@laurent22  Let me know if I'm wrong somewhere and also if any changes needed.

Reply:
But why do you need this shouldUseDarkColors setting value? Can't you read it directly from bridge().shouldUseDarkColors() wherever you need it?

Reply:
>Can't you read it directly from bridge().shouldUseDarkColors() wherever you need it?

I want `bridge().shouldUseDarkColors()` here:
```javascript
if (action.type == 'SETTING_UPDATE_ONE' && action.key === 'preferredDarkTheme' && Setting.value('shouldUseDarkColors')) {
	Setting.setValue('theme', action.value);
}
```
But `bridge()` is not imported in `BaseApplication.js` so created a new setting and set it's value from ConfigScreen.

>But why do you need this shouldUseDarkColors setting value?

We need to check if system detected theme is dark (`bridge().shouldUseDarkColors() == true` ) then only preferred dark theme setting should change `theme` and similarly for light theme.

I don't know if there's any way of calling `shouldUseDarkColors()` function from `BaseApplication.js`

Reply:
Perhaps then in app/app.js, which overrides BaseApplication. I believe it has a middleware too, specific to the desktop app. You can import bridge there

Reply:
Yaa I got it. Thank you sir. Please review the changes, I hope everything is clear now.